### ISSUE TITLE
Ordinal classification [wish list]

### ISSUE THREAD

**Issue by tokestermw** [dmlc/xgboost#695]
Ability to do ordinal classification would be great as no good implementation exists in Python (AFAIK). Training a regression model is a bit slippery on ordinal data (e.g. Amazon product reviews). Might be good for ranking tasks as well.

**Comment by tqchen** [comment]
What is the difference between ordinal classification and ranking? I guess this could be readily implemented as a customized loss function, using some thing like pairwise logistic loss

**Comment by tokestermw** [comment]
Thanks. Also is the `objective='rank:pairwise'` implemented in the Python wrapper?

**Comment by tokestermw** [comment]
Actually, never mind, I think have to use `XGBRegressor(objective="rank:pairwise")` instead of classifier.

    gbm = XGBRegressor(objective="rank:pairwise")

    X =  np.random.normal(0, 1, 100).reshape(10, 10)
    y = np.random.randint(0, 5, 10)
    gbm.fit(X, y)

    x = np.random.normal(0, 1, 100).reshape(10, 10)
    gbm.predict(X)

    # should be in order
    print y[gbm.predict_proba(X)[:, 1].argsort()][::-1]
